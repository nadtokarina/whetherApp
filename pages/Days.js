// На этой странице будет отображен список "избранных " городов, городов, погоду в которых можно посмотреть в быстром доступе
import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity, TextInput} from 'react-native';



export default function Days() {
    return (
        <View style={styles.container}>
            <Text style={styles.title}>Weather</Text>
            <View style={styles.inputWrapper}>
                <TextInput style={styles.textInput} placeholder='Search for the city or the airport...' />
            </View>
            <View style={styles.cityWrapper}>
                <TouchableOpacity style={styles.cities}>
                    <View style={styles.citiNameWrapper}>
                        <Image source={require('../assets/img/rain-icon.png')} style={styles.imgCity}  />
                        <Text style={styles.cityName}>Сургут</Text>
                    </View>
                    <Text style={styles.cityWeather}>+13</Text>
                </TouchableOpacity>
                <TouchableOpacity style={styles.cities}>
                    <View style={styles.citiNameWrapper}>
                        <Image source={require('../assets/img/rain-icon.png')} style={styles.imgCity}  />
                        <Text style={styles.cityName}>Санкт-Петербург</Text>
                    </View>
                    <Text style={styles.cityWeather}>+21</Text>
                </TouchableOpacity>
                <TouchableOpacity style={styles.cities}>
                    <View style={styles.citiNameWrapper}>
                        <Image source={require('../assets/img/wind-icon.png')} style={styles.imgCity} />
                        <Text style={styles.cityName}>Адлер</Text>
                    </View>
                    <Text style={styles.cityWeather}>+29</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}

// Стили 
const styles = StyleSheet.create({
    container: {
        flex: 1,
        paddingTop: 100,
        width: "100%",
        backgroundColor: "#E2A272",
    },
    title: {
        marginTop: -70,
        marginLeft: 10,
        alignSelf: 'right',
        color: "#303345",
        fontSize: 25,
        fontWeight: '700',
    },
    inputWrapper: {
        flexDirection: 'row',
    },
    textInput: {
        margin: 10,
        padding: 3,
        backgroundColor: 'rgba(255, 255, 255, 0.15)',
        width: '95%',
        alignSelf: 'center',
        height: 30,
        borderRadius: 7,
    },
    cityWrapper: {
        // flexDirection: 'row',
    },
    cities: {
        margin: 5,
        width: '85%',
        alignSelf: 'center',
        flexDirection: 'row',
        justifyContent: 'space-between',
        backgroundColor: 'rgba(255, 255, 255, 0.2)',
        padding: 7,
        borderRadius: 15,
        borderColor: 'rgba(255, 255, 255, 0.3)',
        borderWidth: '0.5',
        shadowColor: 'black', 
        shadowOffset: { width: 0, height: 0 }, 
        shadowRadius: 2, 
        shadowOpacity: 0.2,
    },
    cityName: {
        paddingTop: 17,
        fontSize: 16,
        fontWeight: '300',
        color: '#303345',
    },
    cityWeather: {
        paddingTop: 17,
        paddingRight: 15,
        fontSize: 16,
        fontWeight: '500',
        color: '#303345',
    },
    citiNameWrapper: {
        flexDirection: 'row'
    },
    imgCity: {
        marginTop: 5,
    }
})