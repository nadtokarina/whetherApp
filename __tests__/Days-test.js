import React from 'react';
import renderer from 'react-test-renderer';
import Days from '../pages/Days';

test('Days SnapShot', () => {
    const snap = renderer.create(
        <Days /> 
    ).toJSON();

expect(snap).toMatchSnapshot();
});