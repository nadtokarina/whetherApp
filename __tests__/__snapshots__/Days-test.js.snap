// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Days SnapShot 1`] = `
<View
  style={
    {
      "backgroundColor": "#E2A272",
      "flex": 1,
      "paddingTop": 100,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "right",
        "color": "#303345",
        "fontSize": 25,
        "fontWeight": "700",
        "marginLeft": 10,
        "marginTop": -70,
      }
    }
  >
    Weather
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      placeholder="Search for the city or the airport..."
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.15)",
          "borderRadius": 5,
          "height": 30,
          "margin": 10,
          "padding": 3,
          "width": "95%",
        }
      }
    />
  </View>
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "borderColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 15,
          "borderWidth": "0.5",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
          "opacity": 1,
          "padding": 7,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
          "width": "85%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "marginTop": 5,
            }
          }
        />
        <Text
          style={
            {
              "color": "#303345",
              "fontSize": 16,
              "fontWeight": "300",
              "paddingTop": 17,
            }
          }
        >
          Сургут
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#303345",
            "fontSize": 16,
            "fontWeight": "500",
            "paddingRight": 15,
            "paddingTop": 17,
          }
        }
      >
        +13
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "borderColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 15,
          "borderWidth": "0.5",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
          "opacity": 1,
          "padding": 7,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
          "width": "85%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "marginTop": 5,
            }
          }
        />
        <Text
          style={
            {
              "color": "#303345",
              "fontSize": 16,
              "fontWeight": "300",
              "paddingTop": 17,
            }
          }
        >
          Санкт-Петербург
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#303345",
            "fontSize": 16,
            "fontWeight": "500",
            "paddingRight": 15,
            "paddingTop": 17,
          }
        }
      >
        +21
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "borderColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 15,
          "borderWidth": "0.5",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
          "opacity": 1,
          "padding": 7,
          "shadowColor": "black",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
          "width": "85%",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "marginTop": 5,
            }
          }
        />
        <Text
          style={
            {
              "color": "#303345",
              "fontSize": 16,
              "fontWeight": "300",
              "paddingTop": 17,
            }
          }
        >
          Адлер
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#303345",
            "fontSize": 16,
            "fontWeight": "500",
            "paddingRight": 15,
            "paddingTop": 17,
          }
        }
      >
        +29
      </Text>
    </View>
  </View>
</View>
`;
